src/query/service/src/clusters/cluster.rs

```diff
@@ -224,7 +227,18 @@ impl ClusterDiscovery {
             Ok(cluster_nodes) => {
                 let mut res = Vec::with_capacity(cluster_nodes.len());
                 for node in &cluster_nodes {
-                    res.push(Arc::new(node.clone()))
+                    if node.id != self.local_id {
+                        if let Err(cause) = create_client(config, &node.flight_address).await {
+                            warn!(
+                                "Cannot connect node [{:?}], remove it in query. cause: {:?}",
+                                node.flight_address, cause
+                            );
+
+                            continue;
+                        }
+                    }
+
+                    res.push(Arc::new(node.clone()));
                 }
```

Comment:
return an error if res.len() == 0 ?

Reply:
> return an error if res.len() == 0 ?

If `res.len()`, there is only one node is himself.